cli/azd/cmd/show.go

```diff
@@ -0,0 +1,188 @@
+				},
+			}
+
+			res.Services[name] = showSvc
+		}
+
+		// Add information about the target of each service, if we can determine it (if the infrastructure has
+		// not been deployed, for example, we'll just not include target information)
+		resourceManager := infra.NewAzureResourceManager(ctx)
+		if resourceGroupName, err := resourceManager.FindResourceGroupForEnvironment(ctx, &env); err == nil {
+			for name := range prj.Services {
+				if resources, err := project.GetServiceResources(ctx, resourceGroupName, name, &env); err == nil {
+					if len(resources) == 1 {
+						resSvc := res.Services[name]
+						resSvc.Target = &showTargetArm{
+							ResourceId: resources[0].Id,
+						}
+						res.Services[name] = resSvc
+					}
+				} else {
+					log.Printf("ignoring error determining resource id for service %s: %v", name, err)
+				}
+			}
+		} else {
+			log.Printf("ignoring error determining resource group for environment %s, resource ids will not be available: %v", env.GetEnvName(), err)
+		}
+
+		return formatter.Format(res, writer, nil)
+	}
+
+	cmd := commands.Build(
+		commands.ActionFunc(action),
+		rootOptions,
+		"show",
+		"Display information about your application and its resources.",
+		nil,
+	)
+
+	output.AddOutputParam(cmd,
+		[]output.Format{output.JsonFormat, output.NoneFormat},
+		output.NoneFormat,
+	)
+
+	cmd.Hidden = true
+
+	return cmd
+}
+
+type showResult struct {
+	Services map[string]showService `json:"services"`
+}
+
+type showService struct {
+	// Project contains information about the project that backs this service.
+	Project showServiceProject `json:"project"`
+	// Target contains information about the resource that the service is deployed
+	// to.
+	Target *showTargetArm `json:"target,omitempty"`
+}
+
+type showServiceProject struct {
+	// Path contains the path to the project for this service.
+	// When 'type' is 'dotnet', this includes the project file (i.e. Todo.Api.csproj).
+	Path string `json:"path"`
+	// The type of this project. One of "dotnet", "python", or "node"
+	Type string `json:"language"`
+}
+
+type showTargetArm struct {
+	ResourceId string `json:"resourceId"`
+}
+
+func showTypeFromLanguage(language string) string {
+	switch language {
+	case "dotnet":
+		return "dotnet"
+	case "py", "python":
+		return "python"
+	case "ts", "js":
+		return "node"
+	default:
+		panic(fmt.Sprintf("unknown language %s", language))
+	}
+}
+
+// getFullPathToProjectForService returns the full path to the source project for a given service. For dotnet services,
+// this includes the project file (e.g Todo.Api.csproj). For dotnet services, if the `path` component of the configuration
+// does not include the project file, we attempt to determine it by looking for a single .csproj/.vbproj/.fsproj file
+// in that directory. If there are multiple, an error is returned.
+func getFullPathToProjectForService(svc *project.ServiceConfig) (string, error) {
+	if svc.Language == "dotnet" {
```

Comment:
Consider inverting this if statement and performing a short circuit return to reduce the amount of nesting in this func.

Reply:
Yup, makes sense.  Done.